src/SingleArticle.jsx

```diff
@@ -0,0 +1,37 @@
+import { useParams } from 'react-router-dom';
+import { fetchArticleById } from './utils/api';
+import { useEffect, useState } from 'react';
+
+const SingleArticle = () => {
+      const [singleArticle, setSingleArticle] = useState([]);
```

Comment:
`singleArticle` is initialised as an empty array, but you access it as an object. Initialise it as null to align with its usage.